### PR TITLE
Update DB.php

### PR DESCRIPTION
**Pull request by mohnewald**
Added missing $database to mysqli_connect.
Added $dbport to mysqli_connect. (must be set in inc/base/config.php)

 $dbport must be added in the installer, too.

### CODE REVIEW

File: inc/Classes/DB.php [hunk 1]
```diff
@@ -110,11 +110,13 @@ public function connect($save = false)
         $user = $config['database']['user'];
         $pass = $config['database']['passwd'];
         $database = $config['database']['database'];
+        $charset = $config['database']['dbport'];
```
Comment (andygrunwald) [inline inc/Classes/DB.php:113]: Is there a default value if this is not set in the config? `3306` would be a good value.

Comment (M4LuZ) [reply inc/Classes/DB.php:113]: It would even be better if not `$charset` would be overwritten, but `$dbport` ;)

File: inc/Classes/DB.php [hunk 2]
```diff
@@ -110,11 +110,13 @@ public function connect($save = false)
...
         $charset = $config['database']['charset'];
+	    
 
         // Try to connect to the database
         // Suppress error output, because mysqli_connect throws a PHP Warning once it is not able to connect
-        $this->link_id = @mysqli_connect($server, $user, $pass);
+        $this->link_id = @mysqli_connect($server, $user, $pass, $database, $dbport);
```
Comment (M4LuZ) [inline inc/Classes/DB.php:119]: Whilst the code directly below selects the DB as next action I would also be in favour to do so directly on connect.
Not specifying the DB would require the DB user to have `USAGE ON *.*` as privilege, if I'm not mistaken. 
Which _may_ be an issue in shared environments, but _should_ be not.

Just not sure if resulting code does handle DB-selection related errors OK.
Was that tested?

File: inc/Classes/DB.php (hunk 1 above)
Comment (M4LuZ) [reply inc/Classes/DB.php]: resolved with new commit

**Review by andygrunwald — APPROVED** [review]

**Pull request merged**